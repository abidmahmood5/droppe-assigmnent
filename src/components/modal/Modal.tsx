import React, { ReactElement, FunctionComponent } from 'react';
import { FaTimes } from "react-icons/fa";

import Form from "../form";
import {
  ModalCloseButton,
  ModalContent,
  StyledModal,
} from './styles'

import styles from "./modal.module.css";

// TODO: SubmitPayload should be somewhere centralised across components and modules
interface SumbitPayload {
  description: string;
  price: string;
  title: string;
}

interface OwnProps {
  handleSubmit(attrs: SumbitPayload): Promise<void>;
  isOpen: boolean;
  onToggle(): void;
}

type Props = OwnProps;

const Modal: FunctionComponent<Props> = ({
  handleSubmit,
  isOpen,
  onToggle,
}): ReactElement => {
  
  return (
    <StyledModal
      isOpen={isOpen}
      overlayClassName={styles.reactModalOverlay} // classnames package can save me another file
    >
      <ModalContent>
        <ModalCloseButton onClick={onToggle}>
          <FaTimes />
        </ModalCloseButton>

        <Form on-submit={handleSubmit} />
      </ModalContent>
    </StyledModal>
  );
}

export default Modal;