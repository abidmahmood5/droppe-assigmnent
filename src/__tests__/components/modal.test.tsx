import React from 'react';
import renderer from 'react-test-renderer';
import Modal from '../../components/modal';

it('creates a modal component with visibility false', () => {
  const handleSubmit = jest.fn();
  const toggleModalVisibility = jest.fn();

  const modalComponent = renderer.create(
    <Modal
      handleSubmit={handleSubmit}
      isOpen={false}
      onToggle={toggleModalVisibility}
    />
  ).toJSON();

  expect(modalComponent).toMatchSnapshot();
});

