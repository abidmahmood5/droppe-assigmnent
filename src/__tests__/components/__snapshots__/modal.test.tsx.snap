// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a modal component 1`] = `null`;

exports[`creates a modal component with visibility false 1`] = `null`;
